Document runtime dependency resolution and warm-up memoization (#339)
doc-detective/doc-detective: docs/fern/pages/docs/tests/overview.mdx

This keeps your test reporting accurate and prevents skipped tests from incorrectly appearing as passing.

#### Automatic browser installation

> When a context requires a browser that isn't available, Doc Detective tries to install it automatically before skipping. This works for Chrome and Firefox, which can be downloaded on demand along with their WebDriver components (ChromeDriver for Chrome, geckodriver for Firefox).

Reviewer: ```suggestion
When a context requires a browser that isn't available, Doc Detective tries to install it automatically before skipping. This works for Chrome and Firefox, which can be downloaded on demand.
```

> The installation process:
> 
> 1. Detects that the specified browser isn't available
> 2. Downloads and installs the browser and its driver
> 3. Re-checks available browsers
> 4. Proceeds with the test if installation succeeded, or skips if it failed
> 

Reviewer: ```suggestion
```

> Safari and WebKit don't support automatic installation because Safari ships with macOS. If Safari isn't available on your system, Doc Detective skips those contexts without attempting installation.

Reviewer: ```suggestion
Safari doesn't support automatic installation because Safari ships with macOS. If Safari isn't available on your system, Doc Detective skips those contexts without attempting installation.
```

Doc Detective remembers installation outcomes during a test run. If an automatic installation fails for a browser, Doc Detective won't retry that installation for other contexts in the same run.

> #### Driver startup memoization

Reviewer: Drop this while section. The user doesn't need to know about this internal behavior.
